pkg/bootstrap/platform/discovery.go

```diff
@@ -17,16 +17,34 @@ package platform
 import (
 	"sync"
 	"time"
+
+	"istio.io/pkg/env"
+	"istio.io/pkg/log"
 )
 
 const (
 	defaultTimeout = 5 * time.Second
 	numPlatforms   = 3
 )
 
+var CloudPlatform = env.RegisterStringVar("CLOUD_PLATFORM", "", "Clound Platformn on which proxy is running, if not specified,"+
+	"Istio will try to discover the platform. Valid platform vaulues aws,azure,gcp.").Get()
+
 // Discover attempts to discover the host platform, defaulting to
 // `Unknown` if a platform cannot be discovered.
 func Discover() Environment {
+	// First check if user has specified platform - use it if provided.
+	if len(CloudPlatform) > 0 {
+		switch CloudPlatform {
```

Comment:
should we `toLower()` CloudPlatform to allow for `AWS|GCP|Azure` in addition to `aws|gcp|azure` ?